aws/data_source_aws_autoscaling_group_test.go

```diff
@@ -88,8 +107,62 @@ resource "aws_autoscaling_group" "bar" {
   availability_zones        = [data.aws_availability_zones.available.names[0], data.aws_availability_zones.available.names[1]]
 }
 
-data "aws_autoscaling_group" "good_match" {
-  name = aws_autoscaling_group.foo.name
+resource "aws_launch_configuration" "data_source_aws_autoscaling_group_test" {
+  name          = "%[1]s"
+  image_id      = data.aws_ami.ubuntu.id
+  instance_type = "t2.micro"
+}
+
+data "aws_ami" "ubuntu" {
+  most_recent = true
+
+  filter {
+    name   = "name"
+    values = ["ubuntu/images/hvm-ssd/ubuntu-trusty-14.04-amd64-server-*"]
+  }
+
+  filter {
+    name   = "virtualization-type"
+    values = ["hvm"]
+  }
+
+  owners = ["099720109477"] # Canonical
 }
 `, rName)
 }
+
+func testAccAutoScalingGroupDataResourceConfig_launchTemplate() string {
+	return testAccAvailableAZsNoOptInDefaultExcludeConfig() +
+		fmt.Sprintf(`
+data "aws_autoscaling_group" "test" {
+  name = aws_autoscaling_group.test.name
+}
+
+resource "aws_autoscaling_group" "test" {
+  availability_zones = [data.aws_availability_zones.available.names[0]]
+  desired_capacity   = 0
+  max_size           = 0
+  min_size           = 0
+  launch_template {
+    id      = aws_launch_template.test.id
+    version = aws_launch_template.test.default_version
+  }
+}
+
+resource "aws_launch_template" "test" {
+  name_prefix   = "test"
+  image_id      = data.aws_ami.test_ami.id
+  instance_type = "t2.micro"
```

Comment:
not sure if we need to avoid the hardcoded instance type here, but we could also use the `testAccAvailableEc2InstanceTypeForRegion("t3.micro", "t2.micro")` config and ref here